### ISSUE TITLE
ERR Error running script at the moment EmailEngine does partialSync and other software sets Seen flag at the same time 

### ISSUE THREAD

**Issue by archfurry** [postalsys/emailengine#127]
**Describe the bug**
messageNew event is not triggered in case of one (external) system sets Seen flag on message as soon as it comes to IMAP server.
We have currently two systems one old (does not use EmailEngine) and new one with EmailEngine. They are connected to same email account. Old system reads the inbox on timer (every 5 minutes) and sets Seen flag for new messages as they arrive and registered.
EmailEngine at the same time (happens once or twice a day) does partialSync (lib/mailbox.js line 373). But Emailengine uses external library imapFlow and its method [*fetch](https://github.com/postalsys/imapflow/blob/master/lib/imap-flow.js) can process untagged FETCH at the time as regular *fetch executes
So the wrapper function for Lua script at this.entryListGet partialSync (line 399) gets FETCH without uid (or in other words uid is undefined) only MODSEQ fied present. 
After error is thrown new messages previosly fetched in let newMessages = []; are ignored and event is never triggered.

```
[2022-04-18 11:09:28.132 +0000] DEBUG (7 on 03969e50aaed): Initiated IDLE, waiting for server input {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"imap-connection","doneRequested":false}
[2022-04-18 11:09:31.873 +0000] DEBUG (7 on 03969e50aaed): * 106 EXISTS {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"imap-connection","src":"s"}
[2022-04-18 11:09:31.873 +0000] TRACE (7 on 03969e50aaed): Untagged EXISTS {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"mailbox","path":"INBOX","count":106,"prevCount":105}
[2022-04-18 11:09:32.875 +0000] TRACE (7 on 03969e50aaed): Running partial sync {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"mailbox","path":"INBOX"}
[2022-04-18 11:09:32.876 +0000] TRACE (7 on 03969e50aaed): Requesting lock {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"imap-connection","path":"INBOX","lockId":38}
[2022-04-18 11:09:32.876 +0000] TRACE (7 on 03969e50aaed): Mailbox lock acquired {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"imap-connection","path":"INBOX","lockId":38}
[2022-04-18 11:09:32.877 +0000] TRACE (7 on 03969e50aaed): Requesting IDLE break {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"imap-connection","lockId":38,"path":"INBOX","queued":1,"doneRequested":false,"canEnd":true,"doneSent":false}
[2022-04-18 11:09:32.877 +0000] DEBUG (7 on 03969e50aaed): DONE {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"imap-connection","src":"c","comment":"breaking IDLE","lockId":38,"path":"INBOX"}
[2022-04-18 11:09:32.897 +0000] DEBUG (7 on 03969e50aaed): 11F OK Idle completed (4.765 + 4.764 + 4.764 secs). {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"imap-connection","src":"s"}
[2022-04-18 11:09:32.897 +0000] DEBUG (7 on 03969e50aaed): 120 UID FETCH 1:* (UID FLAGS INTERNALDATE MODSEQ) (CHANGEDSINCE 102598) {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"imap-connection","src":"s"}
[2022-04-18 11:09:32.919 +0000] TRACE (7 on 03969e50aaed): * 106 FETCH (UID 115350 FLAGS (\Seen) INTERNALDATE "18-Apr-2022 14:09:31 +0300" MODSEQ (102600)) {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"imap-connection","src":"s"}
[2022-04-18 11:09:32.922 +0000] TRACE (7 on 03969e50aaed): * 106 FETCH (MODSEQ (102600) FLAGS (\Seen)) {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"imap-connection","src":"s"}
[2022-04-18 11:09:32.926 +0000] TRACE (7 on 03969e50aaed): Mailbox lock released {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"imap-connection","lockId":38,"path":"INBOX"}
[2022-04-18 11:09:32.926 +0000] ERROR (7 on 03969e50aaed): ERR Error running script (call to f_cf3f9477aa760eb99fdcbc84714ba8ee40a6cd62): @user_script:6: ERR min or max is not a float {"tid":3,"component":"imap-client","account":"account_1","cid":"8kwd5cpbmzagfossgd1p","sub":"mailbox","path":"INBOX"}
    err: {
      "type": "ReplyError",
      "message": "ERR Error running script (call to f_cf3f9477aa760eb99fdcbc84714ba8ee40a6cd62): @user_script:6: ERR min or max is not a float",
      "stack":
          ReplyError: ERR Error running script (call to f_cf3f9477aa760eb99fdcbc84714ba8ee40a6cd62): @user_script:6: ERR min or max is not a float
              at Mailbox.entryListGet (/emailengine/lib/mailbox.js:143:60)
              at Mailbox.partialSync (/emailengine/lib/mailbox.js:378:52)
              at runMicrotasks (<anonymous>)
              at processTicksAndRejections (node:internal/process/task_queues:96:5)
      "command": {
        "name": "evalsha",
        "args": [
          "cf3f9477aa760eb99fdcbc84714ba8ee40a6cd62",
          "1",
          "iam:account_1:l:pp8vqhpyh90buw26yo4hutu54qvmb1o",
          "NaN"
        ]
      }
    }
```
(possibly) Working fix:
```
for await (let messageData of this.imapClient.fetch(range, fields, opts)) {
                    // ignore Recent flag
                    messageData.flags.delete('\\Recent');
                    
                    if(messageData.uid === undefined){
                            continue;
                    }
                    let storedMessage = await this.entryListGet(messageData.uid, { uid: true });
```
**To Reproduce**
1. Attach EmailEngine account to heavy loaded email with imap "resyncDelay":60
2. run side script (or human script :) ) to set Seen flag on newly arrived messages on that email
3. wait

**Expected behavior**
is to receive new email in any external conditions 

**Enviroment:**
 - OS: Debian 11 Linux debian 5.10.0-9-amd64 #1 SMP Debian 5.10.70-1 (2021-09-30) x86_64 GNU/Linux
 - Docker 
```
Client: Docker Engine - Community
 Version:           20.10.14
 API version:       1.41
 Go version:        go1.16.15
 Git commit:        a224086
 Built:             Thu Mar 24 01:48:21 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.14
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.15
  Git commit:       87a90dc
  Built:            Thu Mar 24 01:46:14 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.5.11
  GitCommit:        3df54a852345ae127d1fa3092b95168e4a88e2f8
 runc:
  Version:          1.0.3
  GitCommit:        v1.0.3-0-gf46b6ba
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

Thank you for your attention!

**Comment by andris9** [comment]
Thank you for the report, this helps a lot. Could you also provide what server/service/software you are using where this issue happens? The server response is totally valid but seems kind of uncommon.

**Comment by archfurry** [comment]
This is mail server or russian dns service. There is not much official info of this service
Our config is 
```
{
  "account": "account_1",
  "name": "account_1",
  "copy": false,
  "notifyFrom": "2022-04-13T00:00:00.000Z",
  "imap": {
    "auth": {
      "user": "redacted@redacted.nichost.ru",
      "pass": "******"
    },
    "host": "mail.nic.ru",
    "port": 993,
    "secure": true,
    "tls": {
      "rejectUnauthorized": true,
      "minVersion": "TLSv1.2"
    },
    "resyncDelay": 60
  },
  "smtp": {
    "auth": {
      "user": "redacted@redacted.nichost.ru",
      "pass": "******"
    },
    "host": "mail.nic.ru",
    "port": 465,
    "secure": true,
    "tls": {
      "rejectUnauthorized": true,
      "minVersion": "TLSv1.2"
    }
  },
  "state": "connected",
  "smtpStatus": {
    "created": "2022-04-19T13:44:06.049Z",
    "status": "ok",
    "response": "250 OK id=1ngo9F-0004Gk-IU"
  },
  "syncTime": "2022-04-19T16:52:37.345Z",
  "lastError": null
}
```

**Comment by andris9** [comment]
Thanks. It's a bug in EmailEngine, but it would be worse if it would regularly happen with a widely used server like Gmail.

**Comment by archfurry** [comment]
Thank you for this great software!

**Comment by andris9** [comment]
Could you try the latest EmailEngine version if it fixes your issue (v2.19.1)? For now, I went with your suggestion to ignore those impartial FETCH responses to avoid uncaught exceptions. For the long term, I need to find a better solution because partial responses are legit, even if uncommon, and the email client should be able to merge partial responses into a single full response.

**Comment by archfurry** [comment]
OK I updated emailenigine to latest version. My fix (as mentioned before) worked two times today. No complains from our test client so far:)
